core-java-modules/core-java-collections-6/src/test/java/com/baeldung/checkedcollections/DataProcessorUnitTest.java

```diff
@@ -0,0 +1,33 @@
+package com.baeldung.checkedcollections;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class DataProcessorUnitTest {
+
+    @Test
+    void givenGenericCollection_whenInvalidTypeDataAdded_failsAfterInvocation() {
+        Collection data = new ArrayList<>();
```

Comment:
We can set the type as Object.
```suggestion
        Collection<Object> data = new ArrayList<>();
```

Reply:
In its companion code, we have
`Collection data = Collections.checkedCollection(new ArrayList<>(), String.class);` and there we cannot use `Collection<Object>`. So I kept `Collection` in both places for sake of consistency.

Reply:
Yes, and that's one point we can mention, we cannot type because we will be breaking the contract and the IDE/syntax checker would complain.

Reply:
Added notes like this into the draft.